src/lib.rs

```diff
@@ -97,3 +99,9 @@ pub trait GenFn4<T>: Fn(&PermutationTable, &Point4<T>) -> T {}
 impl<T, F> GenFn2<T> for F where F: Fn(&PermutationTable, &Point2<T>) -> T, {}
 impl<T, F> GenFn3<T> for F where F: Fn(&PermutationTable, &Point3<T>) -> T, {}
 impl<T, F> GenFn4<T> for F where F: Fn(&PermutationTable, &Point4<T>) -> T, {}
+
+pub trait Module<T> {
+    type Output;
+
+    fn get(&self, point: T) -> Self::Output;
+}
```

Comment:
Actually, with the Clone impl you added to PermutationTable, I can add a #[derive(Copy)] to it and now Perlin can derive Copy as well.

This leads to the question of whether it's better to emphasize using references instead of copy/move, since the Perlin module could take up a decent bit of memory, or more specifically any module that contains a PermutationTable. Long noise chains would inflate the amount of memory used if they use the same PermutationTable-containing module multiple times.